contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+        principal investigator.
         (ii) The identifier enables to link experiments to e.g. proposals.
+      service(NX_CHAR):
+      identifier(NX_CHAR):
+      is_persistent(NX_BOOLEAN):
     experiment_description:
       exists: optional
       doc: |
         An optional free-text description of the experiment.
         
-        However, details of the experiment should be defined in the specific
-        fields of this application definition rather than in this experiment
-        description.
+        Users are strongly advised to parameterize the description of their experiment
+        by using respective groups and fields and base classes instead of writing prose
+        into this field.
+
+        The reason is that such a free-text field is difficult to machine-interpret.
+        The motivation behind keeping this field for now is to learn how far the
+        current base classes need extension based on user feedback.
     experiment_type:
       doc: |
         Specify the type of the optical experiment.
-    start_time(NX_DATE_TIME):
+        
+        Chose other if non of these methods are suitable. You may specify
+        fundamental characteristics or properties in the experimental sub-type.
+      enumeration: [ellipsometry, Raman spectroscopy, photoluminescence, transmission spectroscopy, reflection spectroscopy, other]
+    experiment_sub_type:
+      doc: |
+        Specify a special property or characteristic of the experiment, which specifies
+        the generic experiment type.
+      enumeration: [time resolved, imaging, pump-probe, other]
+    # Replace maybe sub_type with exp_technique?
+    #experimental technique --> pump-probe, parameters seperated (i.e. time, space)?
+    # The above listed things are related to beam properties? This is different
+    # compared to temperature or pressure. Is this a useful distinction?
+    experiment_type_other:
       doc: |
-        Start time of the experiment. UTC offset should be specified.
+        If "other" was selected in "experiment_type", specify the experiment here
+        with a free text name, which is knwon in the respective community of interest.
     (NXuser):
+      exists: [min, 0, max, infty]
       doc: |
-        Contact information of at least the user of the instrument or the
-        investigator who performed this experiment.
-        Adding multiple users, if relevant, is recommended.
+        Contact information and eventually details of at persons who
+        performed the measurements. This can be for example the principal
+        investigator or student.
+        It is recommended to add multiple users if relevant.
+
+        Due to data privacy concerns, there is no minimum requirement.
       name(NX_CHAR):
+        exists: recommended
         doc: |
-          Name of the user.
+          Given (first) name and surname of the user.
       affiliation(NX_CHAR):
```

Comment:
This field (as well as the others) already exits in `NXuser` with virtually the same docstring. Consider removing the optional fields here (as they are anyway part of `NXuser` to shorten the app-def.

Reply:
Ah, yea, maybe I have missed that somehow... Ill as well favor the shorter version and will remove all unnessary parts in NXuser